### PR TITLE
add placing 3D plant feature

### PR DESCRIPTION
**Pull request by AelianaShen**
PR#8 Implement the function of placing plant 3D models on the screen. This feature can be found on the plant detail page and by pressing the AR simulate button.

How to use this feature:
1. go to the **Plants** tab
2. in the recommended list, click a specific plant to open the plant detail page
3. on the detail page, press the **AR simulate** button
4. allow the camera permission if needed
6. follow the AR coach guide to detect the plain
7. tap the screen to place the plant model

[Jira ticket LAB-557](https://oregonstate-innovationlab.atlassian.net/browse/LAB-557)

### CODE REVIEW

**Comment by wbraynen** [comment]
@AelianaShen I don't know how to review this.  There is nothing in your PR description and no link to a ticket.  What's the AC for this?  I could try to review the code blindly, but what am I supposed to be testing and what's this PR supposed to accomplish?

**Comment by AelianaShen** [comment]
> @AelianaShen I don't know how to review this. There is nothing in your PR description and no link to a ticket. What's the AC for this? I could try to review the code blindly, but what am I supposed to be testing and what's this PR supposed to accomplish?

Hi Prof. Will,
Sorry about the mistake. The description has been updated and I also linked to a Jira ticket!

File: PlantApp/Views/ViewContainers/ARPlacePlantViewContainer.swift [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        arView.addCoachingOverlay()
+        
+        return arView
+    }
+    
+    func updateUIView(_ uiView: ARView, context: Context) {}
+    
+    func makeCoordinator() -> PlacePlantCoordinator {
+        return PlacePlantCoordinator(plant: plant)
+    }
```
Comment (wbraynen) [inline PlantApp/Views/ViewContainers/ARPlacePlantViewContainer.swift:38]: `makeCoordinator` isn't called from anywhere.  So, someone else reading this code might think this was left here by mistake.  Maybe add a docstring explaining that you are implementing `https://developer.apple.com/documentation/swiftui/uiviewcontrollerrepresentable/makecoordinator()-32trb` and why?

Same elsewhere where you implement `makeCoordinator`.

File: PlantApp/Models/PlantModel.swift [hunk 2]
```diff
@@ -41,6 +43,7 @@ var plantList = [
         scientificName: "Ficus lyrata 'Bambino'",
         price: "28",
         image: "https://osu-software-innovation-plants-2d-img.s3.us-west-2.amazonaws.com/dwarf_fiddle_leaf_fig.png", //"plant_2",
+        model: "dwarf_fiddle_leaf_fig",
         description: "Mini Fiddle-Leaf Fig or scientifically known as Ficus lyrata 'Bambino,' is a smaller and more compact version of the popular Fiddle-Leaf Fig (Ficus lyrata).",
```
Comment (wbraynen) [inline PlantApp/Models/PlantModel.swift:47]: Can you localize this and other text into at least one other language?  Once you localize it, you will understand why we are not fetching this text from the server (as I think you tried to do earlier).

Localization can be a different ticket, so non-blocking for this PR.

**Review by wbraynen — CHANGES_REQUESTED** [review]
I requested some code comments.  Unfortunately I do not have time to QA this, so, since looks like it's a feature implementation, you'll want to request a review from someone who can QA your work here as you have described in your PR description.  Another option is to have someone QA directly off main, which would help you merge this PR quicker.  Let me know which you prefer.

File: PlantApp/Coordinators/PlacePlantCoordinator.swift [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    init(plant: String) {
+        self.plant = plant
+    }
+    
+    @objc func handleTap(_ reconizer: UITapGestureRecognizer) {
+        guard let arView = arView else { return }
+        let tappedLocation = reconizer.location(in: arView)
+        
+        let modelName = plant //"golden_pothos"
+        let modelEntity = try! Entity.loadModel(named: modelName + ".usdz")
```
[the remark this answers is not in the document]
Comment (AelianaShen) [reply PlantApp/Coordinators/PlacePlantCoordinator.swift:27]: Sounds good! I simply print the message "Failed to load model: \(modelName)" to implement minimal error handling here.

File: PlantApp/Views/ViewContainers/ARPlacePlantViewContainer.swift (hunk 1 above)
Comment (AelianaShen) [reply PlantApp/Views/ViewContainers/ARPlacePlantViewContainer.swift:38]: I believe makeCoordinator was called during the life cycle in `ARPlacePlantViewContainer`. But it's hard for me, this noob, to explain.😳 So, I ask ChatGPT to help me to explain here:

> The `PlacePlantCoordinator` function will be called when the `ARPlacePlantViewContainer` view is created. Let's break down the process:
> 
> 1. When you create an instance of `ARPlacePlantViewContainer`, the `init(plant:)` initializer is called. This initializer sets up the initial state of the `ARPlacePlantViewContainer`, including storing the plant name.
> 
> 2. When SwiftUI needs to create the underlying `UIView` representation of `ARPlacePlantViewContainer`, it calls the `makeUIView(context:)` method. Inside this method:
>    - An `ARView` is created and configured for AR functionality.
>    - A tap gesture recognizer is created and added to the `ARView`.
>    - The `makeCoordinator()` method is called to create an instance of `PlacePlantCoordinator`.
>    - The coordinator's `arView` property is set to the `ARView`.
> 
> 3. The `makeCoordinator()` method creates and returns an instance of `PlacePlantCoordinator`, passing the plant name to its initializer.
> 
> 4. The `handleTap` method of the `PlacePlantCoordinator` instance is set as the action for the tap gesture recognizer.
> 
> 5. The coordinator instance is assigned as the target of the tap gesture recognizer.
> 
> So, in summary, the `PlacePlantCoordinator` function is called when `makeCoordinator()` is invoked during the creation of the `ARPlacePlantViewContainer`. The purpose of `PlacePlantCoordinator` is to handle tap gestures and coordinate actions related to placing plants in the AR scene.
>

**Comment by AelianaShen** [comment]
> I requested some code comments. Unfortunately I do not have time to QA this, so, since looks like it's a feature implementation, you'll want to request a review from someone who can QA your work here as you have described in your PR description. Another option is to have someone QA directly off main, which would help you merge this PR quicker. Let me know which you prefer.

 I prefer to merge first, then invite Radhika to QA the main, if she is willing to help. 🌞

File: PlantApp/Views/ViewContainers/ARPlacePlantViewContainer.swift (hunk 1 above)
Comment (wbraynen) [reply PlantApp/Views/ViewContainers/ARPlacePlantViewContainer.swift:38]: Not sure this helps.  It's a mouthful.  You want to make sure that (a) you understand it yourself and (b) it's more succinct—you just need a sentence or two.

**Comment by wbraynen** [comment]
> > I requested some code comments. Unfortunately I do not have time to QA this, so, since looks like it's a feature implementation, you'll want to request a review from someone who can QA your work here as you have described in your PR description. Another option is to have someone QA directly off main, which would help you merge this PR quicker. Let me know which you prefer.
> 
> I prefer to merge first, then invite Radhika to QA the main, if she is willing to help. 🌞

Also, doesn't have to be Radhika.  You can ask anyone for help, including anyone in the lab but can also be someone else you know.

File: PlantApp/Coordinators/PlacePlantCoordinator.swift (hunk 3 above)
[the remark this answers is not in the document]
Comment (wbraynen) [reply PlantApp/Coordinators/PlacePlantCoordinator.swift]: I mean, that's no error handling at all I think.  You are logging to a console where no one will see it.  But you can think about this later and not high priority.

**Review by wbraynen — APPROVED** [review]
Some nitpicks, nothing blocking.  Approving.

**Pull request merged**